### ISSUE TITLE
linking errors due to lack of `extern` keyword

### ISSUE THREAD

**Issue by etam** [wjaguar/mtPaint#60]
It's required to use `extern` keyword for global variables defined in header files. Otherwise linker fails complaining about symbols existing in multiple object files.

This patch applies to 3.50 release: [mtpaint-3.50-extern.patch.zip](https://github.com/wjaguar/mtPaint/files/6428830/mtpaint-3.50-extern.patch.zip)

**Comment by wjaguar** [comment]
You missed the crucial detail here: required **by specific recent versions of some compilers**.
The configure script is the place where those versions are supposed to be detected, and their intentional violation of half-century of common practice toggled off, by adding the "-fcommon" switch.

If you have a setup where that doesn't work as intended, I'm **very** interested to know what, precisely, is the compiler version where that happens (and knowing the linker and the OS version also won't hurt).

**Comment by etam** [comment]
That explains a lot.

I'm using openSUSE Tumbleweed (rolling release distro). The compiler is gcc 10

```
$ gcc -dumpversion
10
```

I guess that `HAVE_CC_VER` expects a format with major and minor version number.

I run the configure script under `bash -x` and got this:

```
HAVE_GCC_VER 10.0
'[' gcc = gcc ']'
HAVE_CC_VER 10.0
MT_VN0=10
MT_VN1=
MT_VN0=10
MT_VN1=
MT_VN1=
'[' 10 -eq 10 ']'
'[' 0 -le '' ']'
./configure: line 397: [: : integer expression expected
'[' 10 -lt 10 ']'
```
and detection fails.

**Comment by etam** [comment]
This resolves the problem
```patch
Index: mtpaint-3.50/configure
===================================================================
--- mtpaint-3.50.orig/configure
+++ mtpaint-3.50/configure
@@ -389,6 +389,7 @@ HAVE_CC_VER()
 	# First 2 parts of GCC or Clang version
 	MT_VN0=${CCVER%%.*}
 	MT_VN1=${CCVER#$MT_VN0}
+	MT_VN1=${MT_VN1:-0}
 	MT_VN0=${MT_VN0%%[!0-9]*}
 	MT_VN1=${MT_VN1#.}
 	MT_VN1=${MT_VN1%%[!0-9]*}
```

**Comment by wjaguar** [comment]
This was fixed in version 3.50.04
